Objects/unicodeobject.c

```diff
@@ -6109,7 +6109,7 @@ PyUnicode_DecodeUnicodeEscape(const char *s,
     if (result == NULL)
         return NULL;
     if (first_invalid_escape != NULL) {
-        if (PyErr_WarnFormat(PyExc_DeprecationWarning, 1,
+        if (PyErr_WarnFormat(PyExc_SyntaxWarning, 1,
```

Comment:
This is not related to syntax.

Reply:
This warning was added back in 3.6 when we started deprecating the invalid escapes. In 3.8, we're upgrading these to SyntaxWarning so that they're no longer silent by default, and it will eventually be a SyntaxError. The end goal here is to raise a SyntaxError for this use, but this will have to wait for two more releases.

Reply:
This warning is emitted by the "unicode-escape" codec.
```
>>> b'\\z'.decode('unicode-escape')
__main__:1: DeprecationWarning: invalid escape sequence '\z'
'\\z'
```

Reply:
Yes, and this warning [was added][0] as part of bpo-27364 to keep consistency with the rest of the language. It only makes sense to upgrade this warning along with everything else, or else this codec falls out of sync with the rest of the language.

[0]: https://bugs.python.org/file44475/deprecate_invalid_escapes_both_5.patch